pulp_container/app/tasks/sync_stages.py

```diff
@@ -123,13 +127,40 @@ async def run(self):
         ) as pb_parsed_tags:
 
             for download_tag in asyncio.as_completed(to_download):
-                dl_res = await download_tag
-                with open(dl_res.path, "rb") as content_file:
-                    raw_data = content_file.read()
-                dl_res.artifact_attributes["file"] = dl_res.path
-                saved_artifact = await sync_to_async(_save_artifact_blocking)(
-                    dl_res.artifact_attributes
-                )
+                async with ProgressReport(
+                    message="Processing Manifests", code="sync.processing.manifest"
+                ) as pr_manifest:
+                    dl_res = await download_tag
+
+                    digest = dl_res.headers.get("docker-content-digest")
+                    manifest = await sync_to_async(
+                        Manifest.objects.prefetch_related("contentartifact_set")
+                        .filter(digest=digest)
+                        .first
+                    )()
+                    if manifest:
+
+                        def _get_content_data_blocking():
+                            saved_artifact = manifest.contentartifact_set.first().artifact
+                            content_file = saved_artifact.file
+                            raw_data = content_file.read()
```

Comment:
you have a file descriptor leak here file needs to be closed.

Reply:
I am suggesting re-using this https://github.com/pulp/pulp_container/blob/main/pulp_container/app/tasks/sync_stages.py#L431 so we don't have copy/pasted blocks of code in 2 places